honeycombio/resource_trigger.go

```diff
@@ -261,3 +265,48 @@ func flattenTriggerRecipients(rs []honeycombio.TriggerRecipient) []map[string]in
 
 	return result
 }
+
+// Matches read recipients against those declared in HCL
+// and returns the Trigger recipients in a stable order
+// grouped by recipient type.
+//
+// This cannot currently be handled efficiently by a DiffSuppressFunc.
+// See: https://github.com/hashicorp/terraform-plugin-sdk/issues/477
+func matchRecipientsWithSchema(readRecipients []honeycombio.TriggerRecipient, declaredRecipients []interface{}) []honeycombio.TriggerRecipient {
+	result := make([]honeycombio.TriggerRecipient, len(declaredRecipients))
+
+	rMap := make(map[string]honeycombio.TriggerRecipient, len(declaredRecipients))
+	for _, recipient := range readRecipients {
+		rMap[recipient.ID] = recipient
+	}
+
+	for i, declaredRcpt := range declaredRecipients {
+		declaredRcpt := declaredRcpt.(map[string]interface{})
+
+		if declaredRcpt["id"] != "" {
+			if v, ok := rMap[declaredRcpt["id"].(string)]; ok {
+				// matched recipient declared by ID
+				result[i] = v
+				delete(rMap, v.ID)
+			}
+		} else {
+			// group result recipients by type
+			for key, rcpt := range rMap {
+				if string(rcpt.Type) != declaredRcpt["type"] || rcpt.Target != declaredRcpt["target"] {
+					continue
+				}
+
+				result[i] = rcpt
+				delete(rMap, key)
+				break
```

Comment:
I think this would be clearer as a guard instead of a `continue` and _I think_ it's equivalent 🤔 

```suggestion
				if string(rcpt.Type) == declaredRcpt["type"] && rcpt.Target == declaredRcpt["target"] {
					result[i] = rcpt
					delete(rMap, key)
					break
				}
```